pyccel/parser/semantic.py

```diff
@@ -1406,25 +1446,38 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign, *
             # Variable not yet declared (hence array not yet allocated)
             if var is None:
 
+                # Update variable's dictionary with information from function decorators
+                decorators = self._namespace.decorators
+                if decorators:
+                    if 'stack_array' in decorators:
+                        if name in decorators['stack_array']:
+                            d_lhs.update(is_stack_array=True,
+                                    allocatable=False, is_pointer=False, is_target=False)
```

Comment:
Can a stack array be a target? Presumably it shouldn't be the target for anything that will exist beyond the scope of the function, but it should be ok in the function shouldn't it?

Reply:
Good observation, I didn't think of it. Definitely, any automatic variable can be a target. So I should not set `is_target = False` here.

Reply:
Done in 4ddc253